ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -476,3 +522,68 @@ type TracingConstraints blk =
   , Show (Header blk)
   , Show (GenTx blk)
   )
+
+{-------------------------------------------------------------------------------
+  Ancillaries
+-------------------------------------------------------------------------------}
+
+-- | Run @waitAny@ inside several @withAsyc@s
```

Comment:
```suggestion
-- | Spawn multiple async actions and wait for the first one to complete.
```

To my mind, I don't find the original explanation helpful in understanding what this function does!

Reply:
Yes, I was pretty mired in the `async` API when I wrote that. It conveys a lot of details about corner cases (exceptions, interruptions, etc) by mentioning those two "standard" functions. I'll be pushing a commit to elaborate.